test/parsing_spec.rb

```diff
@@ -1,7 +1,178 @@
-require_relative './helper.rb'
-require 'time'
-require 'test/unit'
+require_relative 'spec_helper'
 
+def set_now_to now
+  Timecop.freeze now
+end
+
+describe "parsing strings to get timeframes" do
+
+  let(:parse) { ->(x) { Tickle.parse x } }
+
+  let(:date_matcher) { ->(x, y) { x.to_i.must_equal y.to_i } }
+
+  describe "the basics" do
+
+    [
+      Time.parse('2015-04-27 15:19:14 -0500'),
+      Time.parse('2015-01-01 15:19:14 -0500'),
+      Time.parse('2017-12-25'),
+    ].each do |now|
+
+      [
+        ['day',          now, now + 1.day,   nil, 'day'],
+        ['every day',    now, now + 1.day,   nil, 'day'],
+        ['every week',   now, now + 1.week,  nil, 'week'],
+        ['every month',  now, now + 1.month, nil, 'month'],
+        ['every year',   now, now + 1.year,  nil, 'year'],
+        ###
+        ['daily',        now, now + 1.day,   nil, 'daily'],
+        ['weekly',       now, now + 1.week,  nil, 'weekly'],
+        ['monthly',      now, now + 1.month, nil, 'monthly'],
+        ['yearly',       now, now + 1.year,  nil, 'yearly'],
+        ###
+        ['every 3 days',   now, now + 3.days,   nil,  '3 days'],
+        ['every 3 weeks',  now, now + 3.weeks,  nil,  '3 weeks'],
+        ['every 3 months', now, now + 3.months, nil,  '3 months'],
+        ['every 3 years',  now, now + 3.years,  nil,  '3 years'],
+        ###
+        ['every 9 days',   now, now + 9.days,   nil,  '9 days'],
+        ['every 9 weeks',  now, now + 9.weeks,  nil,  '9 weeks'],
+        ['every 9 months', now, now + 9.months, nil,  '9 months'],
+        ['every 9 years',  now, now + 9.years,  nil,  '9 years'],
+        ###
+        ['every other day',   now, now + 2.days,   nil,  'other day'],
+        ['every other week',  now, now + 2.weeks,  nil,  'other week'],
+        ['every other month', now, now + 2.months, nil,  'other month'],
+        ['every other year',  now, now + 2.years,  nil,  'other year'],
```

Comment:
I added activesupport as a dependency on my branch, because it was my fastest way to do the ```.days``` or ```.weeks``` and that sort of stuff.

I wrote these tests so I could more clearly see the behaviors tied to each text statement.